temporal-sdk/src/main/java/io/temporal/internal/replay/ReplayWorkflowTaskHandler.java

```diff
@@ -168,59 +267,32 @@ private Result failureToResult(PollWorkflowTaskQueueResponse workflowTask, Throw
         workflowType, null, failedRequest.build(), null, null, false);
   }
 
-  private WorkflowRunTaskHandler getOrCreateWorkflowExecutor(
-      PollWorkflowTaskQueueResponse.Builder workflowTask,
-      Scope metricsScope,
-      AtomicBoolean createdNew)
-      throws Exception {
-    WorkflowRunTaskHandler workflowRunTaskHandler;
-    if (stickyTaskQueueName == null) {
-      workflowRunTaskHandler = createStatefulHandler(workflowTask, metricsScope);
-    } else {
-      workflowRunTaskHandler =
-          cache.getOrCreate(
-              workflowTask,
-              metricsScope,
-              () -> {
-                createdNew.set(true);
-                return createStatefulHandler(workflowTask, metricsScope);
-              });
-    }
-    return workflowRunTaskHandler;
-  }
-
-  private Result handleWorkflowTaskWithEmbeddedQuery(
-      PollWorkflowTaskQueueResponse.Builder workflowTask, Scope metricsScope) throws Throwable {
+  /** LEGACY QUERY IMPLEMENTATION */
+  private Result handleQueryOnlyWorkflowTask(
```

Comment:
Honestly I would change this name. It's really not a real workflow task, given we respond on a totally different API

Reply:
While I initially agreed, after a second thought... we still trigger state machines for the workflow task. I would leave the *WorkflowTask name here for this reason.

Reply:
Yeah, I started thinking that after I wrote it. Works for me.